zulip_bots/zulip_bots/bots/chess/chess.py

```diff
@@ -0,0 +1,703 @@
+            '`start as <color> with computer` to start a game.\n\n'
+            'In order to play against a computer, `chess.conf` must be set '
+            'with the key `stockfish_location` set to the location of the '
+            'Stockfish program on this computer.'
+        )
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('chess')
+
+        try:
+            self.engine = chess.uci.popen_engine(
+                self.config_info['stockfish_location']
+            )
+            self.engine.uci()
+        except FileNotFoundError:
+            # It is helpful to allow for fake Stockfish locations if the bot
+            # runner is testing or knows they won't be using an engine.
+            print('That Stockfish doesn\'t exist. Continuing.')
+
+    def handle_message(self, message, bot_handler):
+        START_REGEX = 'start with other user$'
+        START_COMPUTER_REGEX = (
+            'start as (?P<user_color>white|black) with computer'
+        )
+        MOVE_REGEX = 'do (?P<move_san>.+)$'
+        RESIGN_REGEX = 'resign$'
+
+        content = message['content']
+
+        start_regex_match = re.match(START_REGEX, content)
+        start_computer_regex_match = re.match(START_COMPUTER_REGEX, content)
+        move_regex_match = re.match(MOVE_REGEX, content)
+        resign_regex_match = re.match(RESIGN_REGEX, content)
+
+        is_with_computer = bot_handler.storage.get('is_with_computer') if (
+            bot_handler.storage.contains('is_with_computer')
+        ) else False
+
+        last_fen = bot_handler.storage.get('last_fen') if (
+            bot_handler.storage.contains('last_fen')
+        ) else chess.Board().fen()
+
+        if start_regex_match:
+            self.start(message, bot_handler)
+        elif start_computer_regex_match:
+            self.start_computer(
+                message,
+                bot_handler,
+                True if start_computer_regex_match.group(
+                    'user_color'
+                ) == 'white' else False
+            )
+        elif move_regex_match:
+            if is_with_computer:
+                self.move_computer(
+                    message,
+                    bot_handler,
+                    last_fen,
+                    move_regex_match.group('move_san')
+                )
+            else:
+                self.move(
+                    message,
+                    bot_handler,
+                    last_fen,
+                    move_regex_match.group('move_san')
+                )
+        elif resign_regex_match:
+            self.resign(
+                message,
+                bot_handler,
+                last_fen
+            )
+
+    def start(self, message, bot_handler):
+        """Starts a game with another user, with the current user as white.
+        Replies to the bot handler.
+
+        Parameters:
+             - message (object): The Zulip Bots message object.
```

Comment:
We uses Python 3 inline type annotation. (Check out http://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html)

Reply:
@tommyip OK, I added the type annotations in d91b679. Thanks!